src/makeEventList.js

```diff
@@ -230,6 +230,13 @@ function makeEventList(aimsTimeTable, customEventList) {
       const slot = aimsTimeTable.identifiedSlots[index];
       const segmentStartDate = moment(segmentStartDates[startSegment - 1]);
       const segmentEndDate = moment(segmentEndDates[endSegment - 1]);
+      let courseName;
```

Comment:
Nitpick: 
Maybe this can be simplified further by using a single variable. 
To still continue handling the case where `len(identifiedCourseNames) < len(identifiedCourseCodes)`, try using [optionalChaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).